src/main/java/net/minecraftforge/common/extensions/IForgeLivingEntity.java

```diff
@@ -0,0 +1,69 @@
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.common.ForgeMod;
+import net.minecraftforge.fluids.FluidType;
+
+public interface IForgeLivingEntity extends IForgeEntity
+{
+    default LivingEntity self()
+    {
+        return (LivingEntity) this;
+    }
+
+    @Override
+    default boolean canSwimInFluidType(FluidType type)
+    {
+        if (self().isSensitiveToWater() && type == ForgeMod.WATER_TYPE.get()) return false;
+        else return IForgeEntity.super.canSwimInFluidType(type);
+    }
+
+    /**
+     * Performs what to do when an entity attempts to go up or "jump" in a fluid.
+     *
+     * @param type the type of the fluid
+     */
+    default void jumpInFluid(FluidType type)
+    {
+        self().setDeltaMovement(self().getDeltaMovement().add(0.0D, (double)0.04F * self().getAttribute(ForgeMod.SWIM_SPEED.get()).getValue(), 0.0D));
+    }
+
+    /**
+     * Performs what to do when an entity attempts to go down or "sink" in a fluid.
+     *
+     * @param type the type of the fluid
+     */
+    default void sinkInFluid(FluidType type)
+    {
+        self().setDeltaMovement(self().getDeltaMovement().add(0.0D, (double)-0.04F * self().getAttribute(ForgeMod.SWIM_SPEED.get()).getValue(), 0.0D));
+    }
+
+    /**
+     * Returns whether the entity can drown in the fluid.
+     *
+     * @param type the type of the fluid
+     * @return {@code true} if the entity can drown in the fluid, {@code false} otherwise
+     */
+    default boolean canDrownInFluidType(FluidType type)
+    {
+        if (type == ForgeMod.WATER_TYPE.get()) return !self().canBreatheUnderwater();
+        else return type.canDrownIn(self());
```

Comment:
No need for the `else return` this can just be `return` as it is cleaner to read. Also it may be worth considering just making this a ternary expression.